franka_example_controllers/src/cartesian_impedance_example_controller.cpp

```diff
@@ -0,0 +1,249 @@
+    state_interfaces_config.names.push_back(franka_robot_model_name);
+  }
+  // Creates state interface for robot model
+  for (const auto& franka_robot_state_name : franka_robot_state_->get_state_interface_names()) {
+    state_interfaces_config.names.push_back(franka_robot_state_name);
+  }
+  return state_interfaces_config;
+}
+
+CallbackReturn CartesianImpedanceExampleController::on_configure(
+    const rclcpp_lifecycle::State& /*previous_state*/) {
+  arm_id_ = get_node()->get_parameter("arm_id").as_string();
+  franka_robot_state_ = std::make_unique<franka_semantic_components::FrankaRobotState>(
+      franka_semantic_components::FrankaRobotState(arm_id_ + "/" + k_robot_state_interface_name));
+  franka_robot_model_ = std::make_unique<franka_semantic_components::FrankaRobotModel>(
+      franka_semantic_components::FrankaRobotModel(arm_id_ + "/" + k_robot_model_interface_name,
+                                                   arm_id_ + "/" + k_robot_state_interface_name));
+  return CallbackReturn::SUCCESS;
+}
+
+CallbackReturn CartesianImpedanceExampleController::on_activate(
+    const rclcpp_lifecycle::State& /*previous_state*/) {
+  franka_robot_state_->assign_loaned_state_interfaces(state_interfaces_);
+  franka_robot_model_->assign_loaned_state_interfaces(state_interfaces_);
+
+  // compute initial velocity with jacobian and set x_attractor and q_d_nullspace
+  init_robot_state_ = franka_msgs::msg::FrankaRobotState();
+  franka_robot_state_->get_values_as_message(init_robot_state_);
+
+  Eigen::Map<Eigen::Matrix<double, 7, 1>> q_initial(
+      init_robot_state_.measured_joint_state.position.data());
+
+  Eigen::Vector3d position_init(init_robot_state_.o_t_ee.pose.position.x,
+                                init_robot_state_.o_t_ee.pose.position.y,
+                                init_robot_state_.o_t_ee.pose.position.z);
+  Eigen::Quaterniond orientation_init(
+      init_robot_state_.o_t_ee.pose.orientation.w, init_robot_state_.o_t_ee.pose.orientation.x,
+      init_robot_state_.o_t_ee.pose.orientation.y, init_robot_state_.o_t_ee.pose.orientation.z);
+  Eigen::Affine3d initial_transform = Eigen::Affine3d::Identity();
+  initial_transform.translation() = position_init;
+  initial_transform.rotate(orientation_init.toRotationMatrix());
+  position_d_ = initial_transform.translation();
+  orientation_d_ = initial_transform.rotation();
+  position_d_target_ = initial_transform.translation();
+  orientation_d_target_ = initial_transform.rotation();
+
+  q_d_nullspace_ = q_initial;
+  return CallbackReturn::SUCCESS;
+}
+
+controller_interface::CallbackReturn CartesianImpedanceExampleController::on_deactivate(
+    const rclcpp_lifecycle::State& /*previous_state*/) {
+  franka_robot_state_->release_interfaces();
+  franka_robot_model_->release_interfaces();
+  return CallbackReturn::SUCCESS;
+}
+
+controller_interface::return_type CartesianImpedanceExampleController::update(
+    const rclcpp::Time& time,
```

Comment:
comment time and period out to make linter happy

```suggestion

controller_interface::return_type CartesianImpedanceExampleController::update(
    const rclcpp::Time& /*time*/,
    const rclcpp::Duration& /*period*/) {
```